content/docs/patterns/no-series/index.md

```diff
@@ -0,0 +1,186 @@
+        NoSeriesMgt.InitSeries(SalesSetup."Customer Nos.", xRec."No. Series", 0D, "No.", "No. Series");
+    end;
+    [...]
+
+    OnAfterOnInsert(Rec, xRec);
+    end;
+```
+
+In the case of Customer, this is a Data Entity within the Sales module of the system.  The Sales module has a **Sales Setup** table where the user can specify a **No. Series** to use for Customers by default.
+
+`SalesSetup.Get();` fetches the setup table.
+`SalesSetup.TestField("Customer Nos.");` is the basic validation that the **Sales Setup** table has a non-empty **Customer Nos.** field.  If the setup field isn't populated, when the user attempts to create a new Customer, they will receive an error message.
+
+`NoSeriesMgt.InitSeries(SalesSetup."Customer Nos.", xRec."No. Series", 0D, "No.", "No. Series");` is more parameters to a function than most expect.
+
+The function call takes the following parameters:
+
+```AL
+procedure InitSeries(
+  DefaultNoSeriesCode: Code[20]; 
+  OldNoSeriesCode: Code[20];
+  NewDate: Date;
+  var NewNo: Code[20];
+  var NewNoSeriesCode: Code[20])
+```
+
+The **DefaultNoSeriesCode** parameter is typically from a setup table. In the Customer example, this comes from the **Sales Setup** **Customer Nos.** setting.
+
+The **OldNoSeriesCode** is used to verify when changing from one No Series to another that they are related.
+
+The **NewDate** parameter is used to drive numbering based on Dates. This is typically used on Documents.  For master entities, like Customer, an empty date `0D` can be passed in.  
+
+{{% alert title="Note" color="info" %}}
+Many parts of the NoSeriesManagement codeunit predate method overloading, so if the system was created today, some parameters like NewDate would likely be optional.
+{{% /alert %}}
+
+The **NewNo** is a `var` parameter, and is how the new value comes back from the engine.  This also serves two other purposes:
+ - if passed in blank, the Number Series used must be configured to have **Default Nos.** enabled
+ - if passed in with a value, the Number Series used must be configured to have **Manual Nos** enabled.
+
+The **NewNoSeriesCode** is more often used to switch between related number series, but is a required parameter, and is also passed back from the engine, so it is also a `var`.
+
+
+Since the Customer data entity supports the **No. Series Relationship** functionality, there are additional components.  On the table, there is a function called `AssistEdit`:
+
+```AL
+procedure AssistEdit(OldCust: Record Customer): Boolean
+var
+    Cust: Record Customer;
+begin
+    with Cust do begin
+        Cust := Rec;
+        SalesSetup.Get();
+        SalesSetup.TestField("Customer Nos.");
+        if NoSeriesMgt.SelectSeries(SalesSetup."Customer Nos.", OldCust."No. Series", "No. Series") then begin
+            NoSeriesMgt.SetSeries("No.");
+            Rec := Cust;
+            OnAssistEditOnBeforeExit(Cust);
+            exit(true);
+        end;
+    end;
+end;
```

Comment:
`WITH` is being depreciated. Should we really keep this in a newly described pattern?

I know that is from the base app, but still?

Reply:
That one should get a warning tag, that's true.  the WITH is in the Base App, but probably a good idea to call it out as a legacy issue that should not be repeated.